### PR TITLE
Include extra OpenSSL error messages in the trouble shooting section

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
@kacf this is the one we discussed back in early August, now I think it should be in, as we are with OpenSSL full time.

**Comment by merlin-northern** [comment]
@lluiscampos this is something that would be nice to have somewhere around the release. it is not urgent, but we should mention the new errors that can appear which are related to the port to OpenSSL and the security level.

File: 201.Troubleshoot/03.Mender-Client/docs.md [hunk 1]
```diff
@@ -121,6 +121,36 @@ If these mismatch, then you need to update `/etc/mender/server.crt` on your clie
 You can do this manually for testing purposes, and you should
 [include the certificates in your Yocto Project build](../../05.System-updates-Yocto-Project/06.Build-for-production/docs.md#including-the-client-certificates).
 
+## Depth zero self-signed certificate, openssl verify rc: 18
+
+The Mender Client detected a zero depth self-signed certificate, and
+the OpenSSL code cant verify the server identity. You have to either use
```
Comment (kacf) [inline 201.Troubleshoot/03.Mender-Client/docs.md:127]: `s/cant/can't/`

File: 201.Troubleshoot/03.Mender-Client/docs.md [hunk 2]
```diff
@@ -121,6 +121,36 @@ If these mismatch, then you need to update `/etc/mender/server.crt` on your clie
...
+a valid certificate or include the certificate in the local trust store.
```
Comment (kacf) [inline 201.Troubleshoot/03.Mender-Client/docs.md:128]: I'm not really clear on why a zero depth certificate is not a valid certificate. Can you explain a bit more why it isn't valid, and what needs to change for it to become valid?

File: 201.Troubleshoot/03.Mender-Client/docs.md [hunk 3]
```diff
@@ -121,6 +121,36 @@ If these mismatch, then you need to update `/etc/mender/server.crt` on your clie
...
+
+In the above example you can find the configuration file
+at `/opt/local/etc/openssl/openssl.cnf` and check the security level:
+
+```bash
+# cat `openssl version -d | sed -e 's/.*"\([^"]*\)".*/\1/'`/openssl.cnf | grep SECLEVEL
+CipherString = DEFAULT@SECLEVEL=2
+```
+The security level of `2` is the default one on many installations nowadays.
+With the Mender Client ported to OpenSSL it is possible that you see this error
```
Comment (kacf) [inline 201.Troubleshoot/03.Mender-Client/docs.md:151]: Suggestion: "Starting with Mender client 2.4.0, the client uses OpenSSL and it is possible..."

**Review by kacf — CHANGES_REQUESTED** [review]
Good to have this covered!

File: 201.Troubleshoot/03.Mender-Client/docs.md [hunk 4]
```diff
@@ -121,6 +121,36 @@ If these mismatch, then you need to update `/etc/mender/server.crt` on your clie
...
+```
+
+In the above example you can find the configuration file
+at `/opt/local/etc/openssl/openssl.cnf` and check the security level:
+
+```bash
+# cat `openssl version -d | sed -e 's/.*"\([^"]*\)".*/\1/'`/openssl.cnf | grep SECLEVEL
+CipherString = DEFAULT@SECLEVEL=2
+```
+The security level of `2` is the default one on many installations nowadays.
```
Comment (lluiscampos) [inline 201.Troubleshoot/03.Mender-Client/docs.md:150]:
```suggestion
The security level `2` is the default one on many installations nowadays.
```
or
```suggestion
A security level of `2` is the default one on many installations nowadays.
```

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Thanks. I will bring this to 2.5.x release branch

File: 201.Troubleshoot/03.Mender-Client/docs.md (hunk 1 above)
Comment (merlin-northern) [reply 201.Troubleshoot/03.Mender-Client/docs.md]: fixed.

File: 201.Troubleshoot/03.Mender-Client/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 201.Troubleshoot/03.Mender-Client/docs.md]: neither am I. my wife tells me that when I try to explain something no one understands.
fixed. is it better now?

File: 201.Troubleshoot/03.Mender-Client/docs.md (hunk 3 above)
Comment (merlin-northern) [reply 201.Troubleshoot/03.Mender-Client/docs.md]: sure.

File: 201.Troubleshoot/03.Mender-Client/docs.md (hunk 4 above)
Comment (merlin-northern) [reply 201.Troubleshoot/03.Mender-Client/docs.md]: chose second one.

File: 201.Troubleshoot/03.Mender-Client/docs.md (hunk 2 above)
Comment (kacf) [reply 201.Troubleshoot/03.Mender-Client/docs.md]: Yes, that's better, thanks!

**Review by kacf — APPROVED** [review]

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**